website/content/docs/connect/proxies/envoy-extensions/usage/lua.mdx

```diff
@@ -0,0 +1,185 @@
+---
+layout: docs
+page_title: Lua Envoy Extension
+description: >-
+  Learn how the `lua` Envoy extension enables Consul to run Lua scripts during Envoy requests and responses from Consul-generated Envoy resources.
+---
+
+# Run Lua scripts in Envoy proxy
+
+The Lua Envoy extension enables the [HTTP Lua filter](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter) in your Consul Envoy proxies, letting you run Lua scripts when requests and responses pass through Consul-generated Envoy resources.
+
+Envoy filters support setting and getting dynamic metadata, allowing a filter to share state information with subsequent filters. To set dynamic metadata, configure the HTTP Lua Filter. Users can call `streamInfo:dynamicMetadata()` from Lua scripts to get the request's dynamic metadata. 
+
+## Configuration specifications
+
+To use the Lua Envoy extension, configure the following arguments in the `EnvoyExtensions` block:
+
+| Arguments      | Description                                                                                      |
+| -------------- | ------------------------------------------------------------------------------------------------ |
+| `ProxyType`    | Determines the proxy type the extension applies to. The only supported value is `connect-proxy`. |
+| `ListenerType` | Specifies if the extension is applied to the `inbound` or `outbound` listener.                   |
+| `Script`       | The Lua script that is configured to run by the HTTP Lua filter.                                 |
+
+## Workflow
+
+There are two steps to configure the Lua Envoy extension:
+
+1. Configure EnvoyExtensions through `service-defaults` or `proxy-defaults`. 
+1. Apply the configuration entry.
+
+### Configure `EnvoyExtensions`
+
+To use Envoy extensions, you must configure and apply a `proxy-defaults` or `service-defaults` configuration entry with the Envoy extension.
+
+- When you configure Envoy extensions on `proxy-defaults`, they apply to every service.
+- When you configure Envoy extensions on `service-defaults`, they apply to a specific service.
```

Comment:
```suggestion
- When you configure Envoy extensions on `proxy-defaults`, they apply to every service.
- When you configure Envoy extensions on `service-defaults`, they apply to a specific service.
```
It's a little weird to me that these are the only behaviors in a bulleted list. I would consider creating and Introduction section and moving all of the information from line 33 to 39 there. You could even create a subtopic called something like "Configuration entry behaviors" to segment that group of information more specifically. You would still need to introduce this section with something like "Add the `EnvoyExtension` to a `proxy-defaults` or `service-defaults` configuration entry."

Reply:
while i think we can improve on this, for the time being, i'm ok with publishing this as-is

the reason why we opted for the bullet point because it helps with skimming